internal/controller/streams_helper.go

```diff
@@ -124,3 +128,31 @@ func sendResourceSnapshotComplete(sm *streamManager) error {
 	}
 	return sendToResourceStream(sm.logger, sm.streamClient.resourceStream, request)
 }
+
+// sendKeepalive accepts a stream type & sends a keepalive ping on that stream
+func sendKeepalive(sm *streamManager, st StreamType) error {
+	var err error
+
+	switch st {
+	case STREAM_NETWORK_FLOWS:
+		err = sm.streamClient.networkFlowsStream.Send(&pb.SendKubernetesNetworkFlowsRequest{
+			Request: &pb.SendKubernetesNetworkFlowsRequest_Keepalive{},
+		})
+	case STREAM_RESOURCES:
+		err = sm.streamClient.resourceStream.Send(&pb.SendKubernetesResourcesRequest{
+			Request: &pb.SendKubernetesResourcesRequest_Keepalive{},
+		})
+	case STREAM_LOGS:
+		err = sm.streamClient.logStream.Send(&pb.SendLogsRequest{
+			Request: &pb.SendLogsRequest_Keepalive{},
+		})
+	default:
+		return fmt.Errorf("unsupported stream type: %s", st)
+	}
```

Comment:
I doubt this is working without setting the `Keepalive` fields?!!
```suggestion
	switch st {
	case STREAM_NETWORK_FLOWS:
		err = sm.streamClient.networkFlowsStream.Send(&pb.SendKubernetesNetworkFlowsRequest{
			Request: &pb.SendKubernetesNetworkFlowsRequest_Keepalive{
				Keepalive: &pb.Keepalive{},
			},
		})
	case STREAM_RESOURCES:
		err = sm.streamClient.resourceStream.Send(&pb.SendKubernetesResourcesRequest{
			Request: &pb.SendKubernetesResourcesRequest_Keepalive{
				Keepalive: &pb.Keepalive{},
			},
		})
	case STREAM_LOGS:
		err = sm.streamClient.logStream.Send(&pb.SendLogsRequest{
			Request: &pb.SendLogsRequest_Keepalive{
				Keepalive: &pb.Keepalive{},
			},
		})
	default:
		return fmt.Errorf("unsupported stream type: %s", st)
	}
```

Reply:
> And I doubt that the keepalives work without actually setting the Keepalive fields??!! Have you tested that this works without setting them?

We tested it by swapping the Keepalives with empty resource mutations. Just for the resource stream. I will test it properly.